plugins/hls-notes-plugin/src/Ide/Plugin/Notes/Internal.hs

```diff
@@ -0,0 +1,89 @@
+module Ide.Plugin.Notes.Internal where
+
+import           Control.Lens               ((^.))
+import           Control.Monad.IO.Class     (liftIO)
+import qualified Data.Array                 as A
+import           Data.Bifunctor             (first)
+import qualified Data.ByteString.Char8      as BS
+import           Data.Maybe                 (listToMaybe, mapMaybe)
+import qualified Data.Text                  as T
+import qualified Data.Text.Encoding         as TE
+import qualified Data.Text.Encoding.Error   as TE
+import           Development.IDE            (IdeState, Location (..),
+                                             Range (..))
+import           Development.IDE.Core.Rules (getSourceFileSource, runAction)
+import           Ide.PluginUtils            (getNormalizedFilePath,
+                                             pluginResponse, throwPluginError)
+import           Ide.Types                  (PluginMethodHandler)
+import           Language.LSP.Types         (Method (TextDocumentDefinition),
+                                             Position (..), UInt,
+                                             type (|?) (..))
+import qualified Language.LSP.Types.Lens    as L
+import           Prelude                    hiding (lines, mod)
+import           Text.Regex.TDFA            (CompOption (..),
+                                             RegexOptions (defaultExecOpt),
+                                             defaultCompOpt, makeRegexOpts,
+                                             matchAllText)
+import           Text.Regex.TDFA.Text       (Regex)
+
+jumpToNote :: PluginMethodHandler IdeState TextDocumentDefinition
+jumpToNote state _ param = pluginResponse $ do
+    let uriOrig = param ^. (L.textDocument . L.uri)
+    nfp <- getNormalizedFilePath uriOrig
+
+    file <- liftIO $ runAction "Notes.fileContents" state (getSourceFileSource nfp)
```

Comment:
The VFS internally stores things as a rope, which is a line-oriented data structure. You _might_ want to try and get that instead, it would at least let you avoid doing the line splitting yourself.

Reply:
Also I think at least you want `getFileContents`, which gives you `Text` rather than a `ByteString`, which is probably what you should be operating on here!